packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -619,11 +619,13 @@ export interface PluginDeployerHandler {
 }
 
 export const HostedPluginServer = Symbol('HostedPluginServer');
+/**
+ * Plugin metadata is very expensive to load for many plugins, avoid exposing it.
+ */
 export interface HostedPluginServer extends JsonRpcServer<HostedPluginClient> {
 
-    getDeployedMetadata(): Promise<PluginMetadata[]>;
-    getDeployedFrontendMetadata(): Promise<PluginMetadata[]>;
-    getDeployedBackendMetadata(): Promise<PluginMetadata[]>;
+    getDeployedPlugins(): Promise<string[]>;
+    getDeployedPlugin(id: string): Promise<PluginMetadata | undefined>;
```

Comment:
IMHO it looks strange to have same method which is not returning the same object
should be `getDeployedMetadata` for the second

Reply:
Could you elaborate please? Is it about undefined? Do you mean that a plugin cannot be undeployed in the meantime?